src/main/java/org/adridadou/ethereum/EthereumContractInvocationHandler.java

```diff
@@ -114,8 +114,8 @@ private Constructor lookForNonEmptyConstructor(Class<?> returnType, Object[] res
         throw new IllegalArgumentException("no constructor with arguments found! for type " + returnType.getSimpleName());
     }
 
-    <T> void register(T proxy, Class<T> contractInterface, SoliditySource code, String contractName, EthAddress address, EthAccount account) {
-        final Map<String, CompilationResult.ContractMetadata> contractsFound = compile(code.getSource()).contracts;
+    <T> void register(T proxy, Class<T> contractInterface, SoliditySource code, String contractName, EthAddress address, EthAccount account) throws IOException {
+     	final Map<String, CompilationResult.ContractMetadata> contractsFound =  code.isCompiled() ?  CompilationResult.parse(code.getRawJson()).contracts :   compile(code.getSource()).contracts;
```

Comment:
I see your point. It would be interesting to have a look together at StandaloneBlockchain.
I have seen that there is another issue right now as well, you can't define an account when you deploy a smart contract (which would be nice)